scene/resources/packed_scene.cpp

```diff
@@ -175,10 +175,18 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 			ERR_FAIL_COND_V_MSG(n.parent == -1, nullptr, vformat("Invalid scene: node %s does not specify its parent node.", snames[n.name]));
 			NODE_FROM_ID(nparent, n.parent);
 #ifdef DEBUG_ENABLED
-			if (!nparent && (n.parent & FLAG_ID_IS_PATH)) {
-				WARN_PRINT(String("Parent path '" + String(node_paths[n.parent & FLAG_MASK]) + "' for node '" + String(snames[n.name]) + "' has vanished when instantiating: '" + get_path() + "'.").ascii().get_data());
-				old_parent_path = String(node_paths[n.parent & FLAG_MASK]).trim_prefix("./").replace("/", "@");
-				nparent = ret_nodes[0];
+			if (n.owner >= 0) {
+				NODE_FROM_ID(nowner, n.owner);
+				if (nparent && nowner && nowner == ret_nodes[0] && nparent != ret_nodes[0] && nparent->get_scene_file_path().is_empty() && nparent->get_owner() != ret_nodes[0] && !nparent->is_exposed_in_owner() && !nparent->is_child_of_exposed_node()) {
+					WARN_PRINT(String("Exposed parent path '" + ret_nodes[0]->get_path_to(nparent) + "' for node '" + String(snames[n.name]) + "' has vanished when instantiating: '" + get_path() + "'.").ascii().get_data());
```

Comment:
```suggestion
					WARN_PRINT(vformat("Exposed parent path '%s' for node '%s' has vanished when instantiating: '%s'.", ret_nodes[0]->get_path_to(nparent), String(snames[n.name]), get_path()));
```
[Prefer `vformat`](https://docs.godotengine.org/en/latest/contributing/development/core_and_modules/common_engine_methods_and_macros.html#format-a-string) for readability